wiring/src/spark_wiring_ble.cpp

```diff
@@ -328,15 +328,28 @@ hal_ble_uuid_t BleUuid::halUUID() {
 }
 
 uint16_t BleUuid::shorted() const {
-    return uuid_.uuid16;
+    if (type() == BleUuidType::SHORT) {
+        return uuid_.uuid16;
+    } else {
+        return ((uint16_t)uuid_.uuid128[12] | ((uint16_t)uuid_.uuid128[13] << 8));
+    }
 }
 
 void BleUuid::rawBytes(uint8_t uuid128[BLE_SIG_UUID_128BIT_LEN]) const {
```

Comment:
Should this return the number of bytes copied?

Reply:
I was just aware of this. But should it be fixed in this PR? Will do it as long as it makes sense to you.

Reply:
I'm not sure that's why I'm asking :) It just felt weird that we require a `BLE_SIG_UUID_128BIT_LEN` buffer but copy only 2 bytes.

Reply:
I agree. As a workaround, we can add a private member for the `BleUuid` class, which is an 16-bytes length of array to store the 128-bits UUID that is derived from short UUID.

Reply:
I was just going to suggest that.

1. `BleUuid::rawBytes(uint8_t uuid128[BLE_SIG_UUID_128BIT_LEN])` should generate a 128-bit UUID derived from short UUID and copy into the supplied buffer
2. `BleUuid::rawBytes()` should return a pointer to 128-bit UUID derived from short UUID. Ideally we should re-use the same `hal_ble_uuid_t` for storage though, otherwise it seems rather wasteful. This should probably be doable as long as we stick to a convention that `BleUuid` class stores everything as 128-bit UUIDs and perfroms any necessary conversions to/from short form on the boundaries (API and HAL).